libs/common/src/sql-db-entities/payment/payment.entity.test.ts

```diff
@@ -0,0 +1,50 @@
+import { FeeRecordEntityMockBuilder, PaymentEntityMockBuilder, UtilisationReportEntityMockBuilder } from '../../test-helpers';
+
+describe('PaymentEntity', () => {
+  const utilisationReport = UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build();
+
+  describe('updateWithAdditionalFeeRecords', () => {
+    it('adds new fee records to the payment', () => {
+      // Arrange
+      const payment = PaymentEntityMockBuilder.forCurrency('GBP').build();
+
+      const existingFeeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).build();
+      payment.feeRecords = [existingFeeRecord];
+
+      const newFeeRecord1 = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(2).build();
+      const newFeeRecord2 = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(3).build();
+
+      const userId = 'abc123';
+
+      // Act
+      payment.updateWithAdditionalFeeRecords({
+        feeRecords: [newFeeRecord1, newFeeRecord2],
+        requestSource: { platform: 'TFM', userId },
+      });
+
+      // Assert
+      expect(payment.feeRecords).toEqual(expect.arrayContaining([existingFeeRecord, newFeeRecord1, newFeeRecord2]));
+    });
+
+    it('does not allow duplicate fee records to be added to the payment', () => {
```

Comment:
I think we should rename this test as the code does allow us to add two new fee records which are identical which technically is counter to what this test claims it is testing

Reply:
Yep thanks, removed this test

Reply:
Sorry, I think wires got crossed here a bit, I think the test is good just would be better named: `does not allow adding a fee record which is already attached to payment`